[DOCS-10675] Update Heatmaps snapshot user experience in RUM and PA (#28966)
DataDog/documentation: content/en/product_analytics/heatmaps.md

To get started with heatmaps:

1. Verify your SDK version:
   - For Click maps, you must be on the latest version of the SDK (v4.40.0 or later).
   - For Scroll maps, you must be on (v4.50.0 or later).
2. Enable [Session Replay][1].
3. Set`trackUserInteractions: true` in the SDK initialization to enable action tracking (required for Clickmaps).

## Getting started

Navigate to [**Digital Experience > Product Analytics > Session Replay > Heatmaps**][2]. Select your application and view.

From this page, you can select the type of heatmap (Top Elements, Click Map, Scroll Map) you would like to see for a particular view.

{{< img src="product_analytics/heatmaps/pa-heatmaps-page.png" alt="For each view, you can select a different type of heatmap: Top Elements, Click Map, or Scroll Map." style="width:100%;">}}

> Click a view to see a more detailed view of the heatmap. From here, you have the additional view options:

Reviewer: ```suggestion
Click on a view name to have a more detailed look of the heatmap. From here, you have these additional view options:
```
There's a lot of the word `view` in the sentence and referring to slightly different UI items at time? 
The "view" to click on is  referenced as `view name` in the UI, so wanted to keep the consistency. Unless we want to refer to it as `the row of available views`
- "click on a selected row of available view" ?
- or the edit above
- or an edit you think makes more sense

Author: I do like this suggestion, thanks!

- To switch the view being shown, use the **View Name** and **Application** selectors at the top.
- To change the device view, use the **Device type** selector.
- To filter by action name, use the **Filter actions by** dropdown.
- To add more granular filters, like a specific geography for example, click the **Add Filter** button.

{{< img src="product_analytics/heatmaps/pa-heatmaps-annotated.png" alt="The heatmaps page has multiple ways to show different views: by application, map type, device type, action name, and granular filters." style="width:100%;">}}

## Top Elements

> Top Elements heatmaps aggregate click actions on a given view by displaying the most interacted-with elements and their rank. The ranking on the map itself corresponds to the action name on the side.

Reviewer: ```suggestion
Top Elements heatmaps aggregate click actions on a given view by displaying the elements that are most interacted with along with their interaction rank. The ranking on the map itself corresponds to the action name on the side.
```
This worked well but made the modification because of any possible issues with translation for "most interacted-with". If you do not believe this would be an issue, please ignore this edit.

Author: I made a minor edit to your suggestion, but generally agree!

{{< img src="real_user_monitoring/session_replay/heatmaps/heatmaps-top-elements.png" alt="A ranking of the top elements clicked on a page." style="width:100%;">}}

Hover over any action name in the panel to highlight the corresponding action on the map.

## Click maps

> A Click map shows you the most interacted-with actions on a given view by aggregating user click actions from sessions and visualizing them as blobs on the map.

Reviewer: same comment here with "interacted-with" as above.

> A Click map shows you the most interacted-with actions on a given view by aggregating user click actions from sessions and visualizing them as blobs on the map.

Author: ```suggestion
A Click map shows you the actions most interacted with on a given view by aggregating user click actions from sessions and visualizing them as blobs on the map.
```

## Backgrounds

A background is a snapshot of a Session Replay. Each heatmap fetches the 20 backgrounds that triggered the most actions during a given session. Changing the background shows different results depending on the action event selected. 

> **Note**: A heatmap's list of backgrounds cannot be modified.

Author: ```suggestion
```

To select a background snapshot:

> 1. From the heatmap view, click the **Choose Snapshot** button.

Author: ```suggestion
1. From the heatmap view, click the **Change Snapshot** button.
```

   {{< img src="real_user_monitoring/session_replay/heatmaps/heatmaps-change-snapshot-zoomed.png" alt="Click the Change Snapshot button to change the background snapshot the heatmap is based on." style="width:100%;">}}

> 1. Click an action event on the right to select a different background for your heatmap.

Author: ```suggestion
1. Click an action event on the right to select a different snapshot for your heatmap.
```

   {{< img src="real_user_monitoring/session_replay/heatmaps/heatmaps-list-of-events.png" alt="List of action events for the session replay." style="width:100%;">}}

1. If the session does not contain the action that leads to the desired snapshot, you can return to the list of replays by clicking **Choose Another Replay**.
1. Click the **Select Snapshot** button to apply the snapshot at the paused point to the heatmap.

## Next steps

> 

Reviewer 2: current:
### The view that I selected is not showing the initial content.
Heatmaps are generated with Session Replay data. Datadog's intelligent algorithm smartly picks a replay that is both recent and best matches the initial state of the page. In some cases, you might not be able to find the correct replay. To switch the [background](#backgrounds) of your heatmap, you can use the **Choose Background** button to navigate through the different states of the page and find the one you are looking for.

proposed: (also remove the outdated mp4 for now with the old change background UI)
### The view that I selected is not showing the initial content.
Heatmaps are generated with Session Replay data. Datadog’s intelligent algorithm smartly picks a replay that is both recent and best matches the initial state of the page. In some cases, you might not be able to find the correct replay. To switch the [snapshot](#snapshots) of your heatmap, you can use the **Change Snapshot** button to navigate through the different states of a replay and find the one you are looking for. If the replay you’re viewing does not have the snapshot you’re looking for, you can use the **Choose Another Replay** button to select another replay of the same view.

Reviewer 2: current:
### On the action list on the side of my heatmap, I see an icon showing an element that is not visibile in the heatmap.
(image)
The tooltip on the icon says **element is not visible**. This means that the element is a common action on your page, but it's not displayed on the background in the heatmap. To see that element, you can click **Choose Background** in the bottom right corner to switch the background of your heatmap to one where that element is present. 

proposed: (note the 'visible' spelling)
### On the action list on the side of my heatmap, I see an icon showing an element that is not visible in the heatmap.
(image)
The tooltip on the icon says element is not visible. This means that the element is a common action on your page, but it’s not displayed on the snapshot in the heatmap. To see that element, you can click **Change Snapshot** in the top right corner to switch the snapshot of your heatmap to one where that element is present.
